src/Job/Message.php

```diff
@@ -74,11 +81,36 @@ public function getParsedBody(): array
     }
 
     /**
-     * @return mixed
+     * Get a closure containing the callable in the job.
+     *
+     * Supported callables include:
+     *
+     * - strings for global functions, or static method names.
+     * - array of [class, method]. The class will be constructed with no constructor parameters.
+     *
+     * @return \Closure
      */
     public function getCallable()
     {
-        return $this->parsedBody['class'] ?? null;
+        if ($this->callable) {
+            return $this->callable;
+        }
+        $target = $this->parsedBody['class'] ?? null;
+
+        $callable = null;
+        if (is_array($target) && count($target) == 2) {
+            $instance = new $target[0]();
+            $callable = Closure::fromCallable([$instance, $target[1]]);
+        } elseif (is_string($target)) {
+            /** @psalm-suppress InvalidArgument */
+            $callable = Closure::fromCallable($target);
```

Comment:
Why does it think this is an invalid argument?

Reply:
I don't know if it still does, I transported this from its previous home.